code/datums/components/decal.dm

```diff
@@ -0,0 +1,75 @@
+/datum/component/decal
```

Comment:
I've gone ahead and ported over /tg/'s turf_decals, as it makes porting over their maps so much easier (after wasting hours replacing decals manually, so I should've just ported it right away).

Naturally I haven't gone through all of ours maps to replace existing decals. A further PR can handle that.

Reply:
*cough* supermatter port by hand *cough cough*